methcomp/regression.py

```diff
@@ -1,532 +1,279 @@
-        if self.CI is not None and (self.CI > 1 or self.CI < 0):
-            raise ValueError('Confidence interval must be between 0 and 1.')
-        if any([not isinstance(x, str) for x in [self.x_title, self.y_title]]):
-            raise ValueError('Axes labels arguments should be provided as a str.')
-
-    def _derive_params(self):
-        self.n = len(self.method1)
-        self.sv = []
-
-        for i in range(self.n - 1):
-            for j in range(i + 1, self.n):
-                self.sv.append((self.method2[i] - self.method2[j]) /
-                               (self.method1[i] - self.method1[j]))
-
-        self.sv.sort()
-        n = len(self.sv)
-        k = math.floor(len([a for a in self.sv if a < 0]) / 2)
-
-        if n % 2 == 1:
-            self.slope = self.sv[int((n + 1) / 2 + k)]
-        else:
-            self.slope = math.sqrt(self.sv[int(n / 2 + k)] * self.sv[int(n / 2 + k + 1)])
-
-        _ci = st.norm.ppf(1 - (1 - self.CI) / 2) * math.sqrt((self.n * (self.n - 1) * (2 * self.n + 5)) / 18)
-        _m1 = int(round((n - _ci) / 2))
-        _m2 = n - _m1 - 1
-
-        self.slope = [self.slope, self.sv[k + _m1], self.sv[k + _m2]]
-        self.intercept = [np.median(self.method2 - self.slope[0] * self.method1),
-                          np.median(self.method2 - self.slope[1] * self.method1),
-                          np.median(self.method2 - self.slope[2] * self.method1)]
-
-    def plot(self, ax):
-        # plot individual points
-        ax.scatter(self.method1, self.method2, s=20, alpha=0.6, color=self.color_points,
-                   **self.point_kws)
-
-        # plot reference line
-        if self.line_reference:
-            ax.plot([0, 1], [0, 1], label='Reference',
-                    color='grey', linestyle='--', transform=ax.transAxes)
-
-        # plot PaBa-line
-        _xvals = np.array(ax.get_xlim())
-        _yvals = [self.intercept[s] + self.slope[s] * _xvals for s in range(0, 3)]
-        ax.plot(_xvals, _yvals[0], label=f'{self.intercept[0]:.2f} + {self.slope[0]:.2f} * Method 1',
-                color=self.color_paba, linestyle='-')
-        ax.fill_between(_xvals, _yvals[1], _yvals[2], color=self.color_paba, alpha=0.2)
-        if self.line_CI:
-            ax.plot(_xvals, _yvals[1], linestyle='--')
-            ax.plot(_xvals, _yvals[2], linestyle='--')
-
-        if self.legend:
-            ax.legend(loc='upper left', frameon=False)
-
-        ax.set_ylabel(self.y_title)
-        ax.set_xlabel(self.x_title)
-        if self.graph_title is not None:
-            ax.set_title(self.graph_title)
-
-
-def passingbablok(method1, method2,
-                  x_label='Method 1', y_label='Method 2', title=None,
-                  CI=0.95, line_reference=True, line_CI=False, legend=True,
-                  color_points='#000000', color_paba='#008bff',
-                  point_kws=None,
-                  square=False, ax=None):
+    return Deming(
```

Comment:
the deming() function should only return a Deming instance, right? 

the user can then decide whether to calculate parameters, plot, or do both

Reply:
I didn't want to break backwards compatibility. Do you prefer this??
```
deming = Deming(
        method1=method1, method2=method2, CI=CI, vr=vr, sdr=sdr, bootstrap=bootstrap
    )
deming.plot(
        x_label=x_label,
        y_label=y_label,
        title=title,
        line_reference=line_reference,
        line_CI=line_CI,
        legend=legend,
        square=square,
        ax=ax,
        point_kws=point_kws,
        color_regr=color_regr,
        alpha_regr=alpha_regr,
    )
return deming
```

Reply:
Ok I think we can keep it return `.plot()` indeed if we want to keep backwards compatibility. 
I think we should emphasize users to directly use the new classes now, e.g. Deming() instead of the deming() function. That way, we can probably remove the deming, passingbablok, ..., functions in a future version or two. Agreed?

Reply:
Doing this with DeprecationWarning now, using deming, passingbablok, or linear now causes something like this:

```
  /../methcomp/methcomp/methcomp/regression.py:306: DeprecationWarning: methcomp.regression.linear is deprecated, use methcomp.regressor.Linear instead
```